prog/download_boot_image.rb

```diff
@@ -136,8 +136,13 @@ def blob_storage_client
       }.to_json
       sshable.cmd("common/bin/daemonizer 'host/bin/download-boot-image' #{q_daemon_name}", stdin: params_json)
     when "Failed"
-      BootImage.where(vm_host_id: vm_host.id, name: image_name, version: version).destroy
-      fail "Failed to download '#{image_name}' image on #{vm_host}"
+      if !Config.production?
```

Comment:
should we limit the number of retries?

Reply:
We can, but the control plane is designed to retry failures indefinitely. It also has exponential backoff baked in, so it is not defenseless against truly problematic cases. I prefer to keep it as is.

As a tangential note, I added a retry limit for process restarts when respirate crashes, which I saw as a different situation (inside vs. outside of control plane).

Reply:
In the case of `!Config.production?`, we don't raise an exception, so exponential backoff isn't utilized. If there's an issue with the blob storage, it will repeatedly attempt to download and fail until the 40 minutes timeout.

Reply:
Converted `Clog.emit` into a `fail`

Reply:
It won't back off for `!Config.production?` case. If it's `Failed`, it will clean up and raise an exception, increasing the try count. However, in the subsequent run, it will be 'NotStarted' due to the cleanup. It will then run successfully, decreasing the try count. However, if the issue persists, the next run will be `Failed` again. It will repeat until the 40 minutes timeout occurs

Reply:
Ah right... :slightly_smiling_face:

It seems backoff, retry and daemonized runs don't play nicely together. I can try to make them work, but then I'm not sure if we should spend effort there at this point. In development environment, we don't have backoff anyway.

I'm planning to resolve this comment thread and merge the PR after the tests passed.